### DISCUSSION TITLE
Github Actions failure

### DISCUSSION THREAD

**Discussion by willbotics** [cpp-for-yourself/lectures-and-homeworks#54] [Homeworks Q&A] [answered]
Hi - just getting started and followed through from scratch to repeat this issue. From a fresh template of the hw repo, I was unable to successfully run the GitHub actions. Seems to be from the failure to create a progress bar. The rest of the actions do not run.  

What I've tried.
- replaced [your `check-homework.yml`] as described in the CI error msg
- repeated all instructions from the video 

If this is widespread or if I've made some simple mistake I would love to know.
Else I can go deeper into the failed tasks.
CI Screenshot:
![testing_fail](https://user-images.githubusercontent.com/23547000/219225991-b97b0f76-3aa8-4ebb-9545-559869f0f115.png)

**Comment by niosus** [comment, marked as answer]
Thanks for reporting this @Willcampbellmelb. I'll have a look into this. I usually develop these things in the evenings after work and can easily mess these things up 😅 Let's see what this one is about.

**Reply by niosus** [reply]
Just to confirm that I could reproduce this. Now trying to find out why this started happening.

**Reply by niosus** [reply]
There is a notion about something similar happening in the docs for this particular action here: https://github.com/peter-evans/create-or-update-comment#action-inputs

But it refers to forks, not repos created from templates. It might be that GitHub changed this recently 🤔

**Reply by niosus** [reply]
Found it in the docs here: https://docs.github.com/en/actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token

It seems that GitHub by default runs the actions with the read-only access. We would need a write access to make the action work. So that default must be changed.

I could fix the issue you're facing by going into your repository "Settings" then navigate to "Actions" on the left pane, then "General" and choose "Read and write permissions" there. 

This seems to be a new setting that was not there before 🤔 I'm now trying to figure out if there is a way to overcome this to still allow for an automatic setup...

**Reply by niosus** [reply]
@Willcampbellmelb could you try to make the same change to the job in your repo as I did for my small test repo: https://github.com/niosus/new-repo-for-debugging/pull/1/files 

So the whole job file would then read:
```yml
name: CI

on: [pull_request]

jobs:
  run_all_tests:
    name: Run all tests
    uses: cpp-for-yourself/ci-jobs/.github/workflows/check-homework.yml@main
    permissions: 
       contents: write
       pull-requests: write
    with:
      homework-definitions-branch: main
```

Basically, we have to explicitly add write permissions to the job that calls the reusable workflow. Could you try this out and verify for me that it is working?

**Reply by niosus** [reply]
I merged the solution for now so as an alternative to the above you could just create a new repository from a template to verify that it's working for you.

**Reply by willbotics** [reply]
Thanks for looking into this - I can confirm that everything works as it should!
Now onto the course!

**Reply by niosus** [reply]
Thanks for confirming! I've marked this thread as a solution.
